### DISCUSSION TITLE
Paper boundary with restrictTranslate doesn't work after <path> shape is rotated.

### DISCUSSION THREAD

**Discussion by HarryBuxton7** [clientIO/joint#2731] [Q&A] [answered]
### Introduction

If I have paper with a boundary defined by restrictTranslate being set to true, it means my <path> shapes can't go over the boundary, and can be dragged to the edge of the paper as intended.

However when I rotate my items using the halo rotate tool, I can't drag the item to the edge of the paper, there is invisible whitespace blocking it. I assume because the bounding box has not been updated.

The shape being along the edge:

![image](https://github.com/user-attachments/assets/bb7fd56b-2fcd-4ff1-a238-6c7655408361)

Once rotated, it can no longer be dragged to the edge:

![image](https://github.com/user-attachments/assets/9b62fb40-b905-4ee8-a0b2-ef3395fe3f83)

### Steps to reproduce

[Codesandbox](https://codesandbox.io/p/sandbox/lucid-cherry-k8ww57?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clzjvannc0007356ilspcq829%2522%252C%2522sizes%2522%253A%255B100%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clzjvannb0003356i5ircpjh7%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clzjvannb0004356i2n0gfzb0%2522%257D%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clzjvannc0006356ifh91dcqo%2522%257D%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clzjvannb0003356i5ircpjh7%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clzjvannb0002356iu7zphub8%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A2%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A2%252C%2522endColumn%2522%253A1%257D%255D%257D%255D%252C%2522id%2522%253A%2522clzjvannb0003356i5ircpjh7%2522%252C%2522activeTabId%2522%253A%2522clzjvannb0002356iu7zphub8%2522%257D%252C%2522clzjvannc0006356ifh91dcqo%2522%253A%257B%2522id%2522%253A%2522clzjvannc0006356ifh91dcqo%2522%252C%2522activeTabId%2522%253A%2522clzjvcfrk000x356ijukngkve%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clzjvannb0005356iv5iooxuj%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clzjvcfrk000x356ijukngkve%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clzjvannb0004356i2n0gfzb0%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clzjvannb0004356i2n0gfzb0%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Restrictions & Constraints

I need to be using a custom <path> shape and have a fixed x by x boundary I can drag to the side of.

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
It works as intended. We take only the rotated model bounding box into account which is always a rectangle.
<img width="324" alt="image" src="https://github.com/user-attachments/assets/3cc77eff-05d0-41f8-a86b-b068fc3e1446">
_`blue` is the element bounding box and `red` is rotated element bounding box_

The `restrictTranslate()` could be defined as a function returning a function, if you need to implement a custom logic.
Here's an example: https://www.jointjs.com/demos/hexagonal-grid
